ovn-tester/ovn_workload.py

```diff
@@ -140,21 +142,35 @@ def configure_localnet(self, physical_net):
         self.run(cmd=f'ovs-vsctl -- set open_vswitch . '
                  f'external-ids:ovn-bridge-mappings={physical_net}:br-ex')
 
+    def configure_external_host_net_str(self, ip, gw):
+        cmd = ''
+        if ip.get_ip4():
+            cmd += f'ip -n ext-ns addr add {ip.get_ip4()}/{ip.get_plen4()}' \
+                   f' dev veth0 ; '
+        if gw.get_ip4():
+            cmd += f'ip -n ext-ns route add default via {gw.get_ip4()} ; '
+
+        if ip.get_ip6():
+            cmd += f'ip -n ext-ns addr add {ip.get_ip6()}/{ip.get_plen6()}' \
+                   f' dev veth0 ; '
+        if gw.get_ip6():
+            cmd += f'ip -n ext-ns route add default via {gw.get_ip6()} ; '
+        return cmd
+
     def configure_external_host(self):
         log.info(f'Adding external host on {self.container}')
-        gw_ip = netaddr.IPAddress(self.ext_net.last - 1)
-        host_ip = netaddr.IPAddress(self.ext_net.last - 2)
-
-        self.run(cmd='ip link add veth0 type veth peer name veth1')
-        self.run(cmd='ip netns add ext-ns')
-        self.run(cmd='ip link set netns ext-ns dev veth0')
-        self.run(cmd='ip netns exec ext-ns ip link set dev veth0 up')
-        self.run(cmd=f'ip netns exec ext-ns '
-                 f'ip addr add {host_ip}/{self.ext_net.prefixlen} '
-                 f'dev veth0')
-        self.run(cmd=f'ip netns exec ext-ns ip route add default via {gw_ip}')
-        self.run(cmd='ip link set dev veth1 up')
-        self.run(cmd='ovs-vsctl add-port br-ex veth1')
+
+        cmd = 'bash -c \'ip link add veth0 type veth peer name veth1 ; ' \
+              'ip netns add ext-ns ; ' \
+              'ip link set netns ext-ns dev veth0 ; ' \
+              'ip -n ext-ns link set dev veth0 up ; '
+        cmd += self.configure_external_host_net_str(
+                self.ext_net.reverse(2),
+                self.ext_net.reverse()
+            )
+        cmd += 'ip link set dev veth1 up ; ' \
+               'ovs-vsctl add-port br-ex veth1 \''
```

Comment:
Nit: I don't really like the "imbalance".  Here we add the last `'`.  We don't care about performance that much, we could change this to:
```
    def configure_external_host(self):
        log.info(f'Adding external host on {self.container}')
        cmd = 'ip link add veth0 type veth peer name veth1 ; ' \
              'ip netns add ext-ns ; ' \
              'ip link set netns ext-ns dev veth0 ; ' \
              'ip -n ext-ns link set dev veth0 up ; '
        cmd += self.configure_external_host_net_str(
                self.ext_net.reverse(2),
                self.ext_net.reverse()
            )
        cmd += 'ip link set dev veth1 up ; ' \
               'ovs-vsctl add-port br-ex veth1'

        # Run as a single invocation:
        cmd = f'bash -c \'{cmd}\''
        self.run(cmd=cmd)
```

Reply:
@dceara ack, I will fix it